scheduling/tasks.go

```diff
@@ -0,0 +1,120 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskId int32 // need to determine how we want to create IDs
+type RunId int32  // need to determine how we want to create IDs
+type TaskType string
+
+const (
+	ResourceCreation TaskType = "ResourceCreation"
+	HealthCheck      TaskType = "HealthCheck"
+	Monitoring       TaskType = "Monitoring"
+)
+
+type TargetType string
+
+const (
+	ProviderTarget    TargetType = "Provider"
+	NameVariantTarget TargetType = "NameVariant"
+)
+
+type Provider struct {
+	Name       string     `json:"name"`
+	TargetType TargetType `json:"targetType"`
+}
+
+type NameVariant struct {
+	Name       string     `json:"name"`
+	TargetType TargetType `json:"targetType"`
+}
+
+func (p Provider) Type() TargetType {
+	return p.TargetType
+}
+
+func (nv NameVariant) Type() TargetType {
+	return nv.TargetType
+}
+
+type TaskTarget interface {
+	Type() TargetType
+}
+
+type TaskMetadata struct {
+	ID          TaskId     `json:"id"`
+	Name        string     `json:"name"`
+	TaskType    TaskType   `json:"type"`
+	Target      TaskTarget `json:"target"`
+	DateCreated time.Time  `json:"dateCreated"`
+}
+
+func (t *TaskMetadata) ToJSON() ([]byte, error) {
+	return json.Marshal(t)
+}
+
+func (t *TaskMetadata) FromJSON(data []byte) error {
```

Comment:
same thing about json naming, should we use traditional json naming methods

Reply:
I think @sdreyer wanted to stuck to FromJSON and ToJson

Reply:
why? lol

"Using MarshalJSON and UnmarshalJSON is idiomatic when implementing custom JSON marshaling and unmarshaling behavior. These names are directly inspired by the standard library and are especially used when implementing the json.Marshaler and json.Unmarshaler interfaces"

Reply:
we can use marshal and unmarshal. I dont have a strong opinion